### PR TITLE
Only output build errors, not warnings, on error

### PR DESCRIPTION
**Pull request by guybedford**
This resolves https://github.com/zeit/ncc/issues/373, so that TypeScript errors will only display the direct error message, instead of all the possible Webpack warnings first.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -133,7 +133,8 @@ module.exports = (
         ))
           return callback();
       } while (pkgPath.length > (pkgPath = pkgPath.substr(0, pkgPath.lastIndexOf(sep, pkgPath.length - 15 - request.length)) + sep + 'node_modules' + sep + request).length);
-      console.error(`ncc: Module directory "${context}" attempted to require "${request}" but could not be resolved, assuming external.`);
+      if (debugLog)
+        console.error(`ncc: Module directory "${context}" attempted to require "${request}" but could not be resolved, assuming external.`);
```
Comment (styfle) [inline src/index.js]: I'm torn on this one. In production builds, its nice to know what was assumed as external. On the other hand, I don't think I have seen and bugs caused by the assumption of `external`

File: test/cli.js [hunk 2]
```diff
@@ -31,9 +31,15 @@
     expect: { timeout: true }
   },
   {
-    args: ["run", "test/fixtures/fail.ts"],
+    args: ["run", "test/fixtures/ts-error1/fail.ts"],
```
Comment (styfle) [inline test/cli.js]: Why was this changed?

File: src/index.js (hunk 1 above)
Comment (guybedford) [reply src/index.js]: Ok, if we're not sure lets leave it in for now.

File: test/cli.js (hunk 2 above)
Comment (guybedford) [reply test/cli.js]: Because ts-loader will throw errors for all ts files in the project, such that both error messages become the error message for the one file otherwise!

I kind of see this as something we should address in https://github.com/zeit/ncc/issues/362.

File: src/index.js [hunk 3]
```diff
@@ -217,8 +217,13 @@ module.exports = (
         if (err) return reject(err);
         compiler.close(err => {
           if (err) return reject(err);
-          if (stats.hasErrors())
-            return reject(new Error(stats.toString()));
+          if (stats.hasErrors()) {
+            let errLog = '';
+            stats.compilation.errors.forEach((err, index) => {
```
Comment (styfle) [inline src/index.js]: This is an interesting way to do it. I would have done

```
stats.compilation.errors.map(err => err.message).join('\n')
```

But I guess yours only loops once?

Comment (guybedford) [reply src/index.js]: Sure, let's keep it simple.

**Comment by styfle** [comment]
@guybedford CI is failing though

**Comment by guybedford** [comment]
Just figuring out why CI is failing... should be able to get this green shortly.

**Comment by guybedford** [comment]
Ok, should pass this run with some luck.
